### PR TITLE
CLOUD-818 delete unused users

### PR DESCRIPTION
**Pull request by nmarukovich**

### CODE REVIEW

File: cloud/aws-functions/orphaned_openshift_users.py [hunk 1]
```diff
@@ -0,0 +1,70 @@
...
+
+def delete_user(client, user_name):
+    try:
+        client.delete_user(UserName=user_name)
+    except client.exceptions.NoSuchEntityException as e:
+        logging.error(f"Delete of user failed with error: {e}")
+
+def delete_user_policies(client, user_name):
+    user_policies = client.list_user_policies(UserName=user_name)['PolicyNames']
+    print(user_policies)
```
Comment (egegunes) [inline cloud/aws-functions/orphaned_openshift_users.py:47]: this looks like a leftover from debug

File: cloud/aws-functions/orphaned_openshift_users.py [hunk 2]
```diff
@@ -0,0 +1,70 @@
...
+    if user_policies:
+        for policy in user_policies:
+            client.delete_user_policy(UserName=user_name, PolicyName=policy)
+
+def delete_user_access_keys(client, user_name):
+    user_access_keys = client.list_access_keys(UserName=user_name)['AccessKeyMetadata']
+    print(f"{user_name} {user_access_keys}")
```
Comment (egegunes) [inline cloud/aws-functions/orphaned_openshift_users.py:54]: probably don't want to print these

**Review by egegunes — CHANGES_REQUESTED** [review]

File: cloud/aws-functions/orphaned_openshift_users.py (hunk 2 above)
Comment (nmarukovich) [reply cloud/aws-functions/orphaned_openshift_users.py:54]: yes! sorry)

File: cloud/aws-functions/orphaned_openshift_users.py [hunk 3]
```diff
@@ -0,0 +1,67 @@
...
+
+
+def delete_user(client, user_name):
+    try:
+        client.delete_user(UserName=user_name)
+    except client.exceptions.NoSuchEntityException as e:
+        logging.error(f"Delete of user failed with error: {e}")
+
+def delete_user_policies(client, user_name):
+    user_policies = client.list_user_policies(UserName=user_name)['PolicyNames']
```
Comment (egegunes) [inline cloud/aws-functions/orphaned_openshift_users.py:45]: do these keys always exist in the return value but can be `None`?

File: cloud/aws-functions/orphaned_openshift_users.py [hunk 4]
```diff
@@ -0,0 +1,67 @@
...
+    if user_policies:
+        for policy in user_policies:
+            client.delete_user_policy(UserName=user_name, PolicyName=policy)
+
+def delete_user_access_keys(client, user_name):
+    user_access_keys = client.list_access_keys(UserName=user_name)['AccessKeyMetadata']
```
Comment (egegunes) [inline cloud/aws-functions/orphaned_openshift_users.py:51]: do these keys always exist in the return value but can be `None`?

Comment (nmarukovich) [reply cloud/aws-functions/orphaned_openshift_users.py:51]: the same. if not we return empty list.

File: cloud/aws-functions/orphaned_openshift_users.py [hunk 5]
```diff
@@ -0,0 +1,67 @@
...
+    for key in tags_dict.keys():
+        if 'kubernetes.io/cluster/openshift' not in key:
+            return False
+
+    current_time = datetime.datetime.now().timestamp()
+
+    creation_time = int(user['CreateDate'].timestamp())
+    if (current_time - creation_time) / 3600 > 24:
+        return True
+    return False
```
[the remark this answers is not in the document]
Comment (nmarukovich) [reply cloud/aws-functions/orphaned_openshift_users.py:22]: This is super nice! I like it!

**Review by egegunes — APPROVED** [review]

**Review by hors — APPROVED** [review]

**Pull request merged**